ArjunaJTS/jtax/tests/classes/com/arjuna/ats/internal/jta/recovery/jts/JCAServerTransactionRecoveryModuleTest.java

```diff
@@ -0,0 +1,155 @@
+        ORB myORB = ORB.getInstance("test");
+        RootOA myOA = OA.getRootOA(myORB);
+
+        myORB.initORB(new String[0], null);
+        myOA.initOA();
+
+        com.arjuna.ats.internal.jts.ORBManager.setORB(myORB);
+        com.arjuna.ats.internal.jts.ORBManager.setPOA(myOA);
+
+        RecoveryManager.manager().initialize();
+        Implementationsx.initialise();
+    }
+
+    @After
+    public void tearDown() {
+        RecoveryManager.manager().terminate();
+
+        ORB myORB = ORB.getInstance("test");
+        RootOA myOA = OA.getRootOA(myORB);
+        myOA.destroy();
+        myORB.shutdown();
+    }
+
+    @Test
+    public void testIdempotentLoad() throws XAException, RollbackException, SystemException {
+        JCAServerTransactionRecoveryModule module = new JCAServerTransactionRecoveryModule();
+        XATerminator terminator = new XATerminatorImple();
+        Xid xid = new Xid() {
+
+            @Override
+            public int getFormatId() {
+                return 0;
+            }
+
+            @Override
+            public byte[] getGlobalTransactionId() {
+                return new byte[0];
+            }
+
+            @Override
+            public byte[] getBranchQualifier() {
+                return new byte[0];
+            }
+        };
+        SubordinationManager.getTransactionImporter().importTransaction(xid);
+        SubordinateTransaction importedTransaction = SubordinationManager.getTransactionImporter().getImportedTransaction(xid);
+        importedTransaction.enlistResource(new XAResource() {
+            @Override
+            public void commit(Xid xid, boolean b) throws XAException {
+                commitCalled = true;
+            }
+
+            @Override
+            public void end(Xid xid, int i) throws XAException {
+
+            }
+
+            @Override
+            public void forget(Xid xid) throws XAException {
+
+            }
+
+            @Override
+            public int getTransactionTimeout() throws XAException {
+                return 0;
+            }
+
+            @Override
+            public boolean isSameRM(XAResource xaResource) throws XAException {
+                return false;
+            }
+
+            @Override
+            public int prepare(Xid xid) throws XAException {
+                return 0;
+            }
+
+            @Override
+            public Xid[] recover(int i) throws XAException {
+                return new Xid[0];
+            }
+
+            @Override
+            public void rollback(Xid xid) throws XAException {
+
+            }
+
+            @Override
+            public boolean setTransactionTimeout(int i) throws XAException {
+                return false;
+            }
+
+            @Override
+            public void start(Xid xid, int i) throws XAException {
+
+            }
+        });
+        terminator.prepare(xid);
+        module.periodicWorkFirstPass();
+        module.periodicWorkFirstPass();
```

Comment:
Aha, hence the name of the test. I am not sure why you feel that warrants a test but I will mark the pr as ready to merge anyway.